src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -1207,8 +1208,7 @@ private void setupGCSClient(
         // We are authenticated with an oauth access token.
         this.gcsClient =
             StorageOptions.newBuilder()
-                .setHeaderProvider(
```

Comment:
I suggest, keep these lines to set credentials in headers also, as this was working for all most for all cases. GCS is gone through many testing and keeping these line will make sure it won't have any regression. We add new lines to set credentials for environment where they have default credentials sets.

Reply:
we keep this line